src/hook/useInputs.tsx

```diff
@@ -0,0 +1,109 @@
+import { ChangeEvent, useRef, useState, KeyboardEvent } from 'react';
+
+export const createUseInputConfig = (maxLength?: number, pattern?: RegExp) => ({ maxLength, pattern });
+const INPUT_CONFIG_DEFAULTS = [createUseInputConfig()];
+
+type CursorPosition = 'start' | 'end';
+
+type InputConfig = {
+  maxLength?: number;
+  pattern?: RegExp;
+};
+type UseInputsProps = InputConfig[];
+
+export const useInputs = (inputConfigs: UseInputsProps = INPUT_CONFIG_DEFAULTS) => {
+  const INITIAL_VALUES = inputConfigs.map(() => '');
+  const [values, setValues] = useState(INITIAL_VALUES);
+  const refs = inputConfigs.map(() => useRef<HTMLInputElement>(null));
+
+  const createChangeHandlerByIndex = (index: number) => (event: ChangeEvent<HTMLInputElement>) => {
+    const { value: inputValue } = event.target;
+    const config = inputConfigs[index];
+    const isValid = !config.pattern || config.pattern.test(inputValue);
+
+    if (isValid && (!config.maxLength || inputValue.length <= config.maxLength)) {
+      const inputValues = [...values];
+      inputValues[index] = inputValue;
+      setValues(inputValues);
```

Comment:
```suggestion
      inputValues[index] = inputValue;
      setValues(inputValues);
```
이렇게 표현해도 무방하도록 seValues 에서 명시해주면 어떨까 싶기도 하네요.
혹은 이미 팀의 정책이라면 어쩔 수 없고..?

Reply:
> 이렇게 표현해도 무방하도록 seValues 에서 명시해주면 어떨까 싶기도 하네요. 혹은 이미 팀의 정책이라면 어쩔 수 없고..?

useState 불변성 유지를 위해서 이렇게 했습니다.
```
  const [values, setValues] = useState(INITIAL_VALUES);

  ...

  const inputValues = [...values];
  inputValues[index] = inputValue;
  setValues(inputValues);
```